testplan/web_ui/web_app.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+"""
+Web application for Testplan & Monitor UIs,
+"""
+from testplan.common.utils.path import pwd
+from flask import Flask, send_from_directory, abort
+from flask_restplus import Resource, Api
+from werkzeug.exceptions import NotFound, NotImplemented
+import argparse
+import os
+
+app = Flask(__name__)
+_api = Api(app)
+
+def parse_cli_args():
+    """Web App command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--static-path', nargs='?', default=None, const=pwd())
+    parser.add_argument('--data-path', nargs='?', default=None, const=pwd())
+    return parser.parse_args()
+
+@_api.route('/testplan/<string:report_uid>')
+class Testplan(Resource):
+    def get(self, report_uid):
+        """Get a Testplan report (HTML) given it's uid."""
```

Comment:
Do we want to validate that the UID exists here (and return 404 if not), or will the web app handle that?

Reply:
No as explained above this endpoint doesn't actually make use of the UID, therefore I don't think it should validate it. At most it would just inject the UID into the HTML page. The HTML page would then use this UID to make a /testplan/<uid>/report request which can respond with either the report.json or a 404 (validating the JSON). The HTML page then knows how to handle this and can show an error if a 404 is received.